osquery/tables/system/windows/system_info.cpp

```diff
@@ -39,8 +39,11 @@ QueryData genSystemInfo(QueryContext& context) {
   const auto wmiSystemReq =
       WmiRequest::CreateWmiRequest("select * from Win32_ComputerSystem");
   auto wmiExecutedSuccessful = wmiSystemReq.isValue();
+  // Why not select * here?  Because we only use NumberOfCores out of this structure,
+  // however getting the full structure takes a 1s per CPU core penalty, which really
+  // hurts on large CPU systems
   const auto wmiSystemReqProc =
-      WmiRequest::CreateWmiRequest("select * from Win32_Processor");
+      WmiRequest::CreateWmiRequest("select NumberOfCores from Win32_Processor");
```

Comment:
I see `NumberOfLogicalProcessors`, `NumberOfProcessors`, and `NumberOfCores` listed below, so I'm not convinced this works. Do we maybe need to select all 3?

Reply:
The other two come from the other WMI query `select * from Win32_ComputerSystem` - I was originally confused by this too.  

THere is wmiResults, which comes from the first query (computersystem) which is fast, and wmiResultsProc , which comes from the second query and is only used by NumberOfCores

Reply:
Aha! Thank you for the clarification :)